### PR TITLE
pkg: update mac brew pkg to use combined cask/formula commands

### PR DESCRIPTION
**Pull request by SMillerDev**
### What issues does this PR fix or reference?
Fixes: improves interaction with Homebrew package manager

### Previous Behavior
Specific to macOS and no usage of shared commands for cask and formula

### New Behavior
Allow Homebrew anywhere and optimize commands

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by Ch3LL** [comment]
Does this PR https://github.com/saltstack/salt/pull/59439/files implement your fix?

**Comment by SMillerDev** [comment]
It seems there is some overlap yes, I'll rebase.

**Comment by SMillerDev** [comment]
Took a while, but I finally rebased it

File: salt/modules/mac_brew_pkg.py [hunk 1]
```diff
@@ -350,6 +348,7 @@ def _info(*pkgs):
     if brew_result["retcode"]:
         log.error("Failed to get info about packages: %s", " ".join(pkgs))
         return {}
+    ret = {}
```
Comment (Ch3LL) [inline salt/modules/mac_brew_pkg.py:351]: I don't think this is needed here?

**Review by Ch3LL — CHANGES_REQUESTED** [review]
this will also require a changelog

**Comment by SMillerDev** [comment]
> this will also require a changelog

@Ch3LL where/how do I write that?

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**